packages/block-library/src/gallery/gallery.native.js

```diff
@@ -90,7 +92,7 @@ export const Gallery = ( props ) => {
 							key={ img.id || img.url }
 							url={ img.url }
 							alt={ img.alt }
-							id={ img.id }
+							id={ parseInt( img.id ) } // make id an integer explicitly
```

Comment:
What do you think if we specify a radix for this `parseInt`? like

```suggestion
							id={ parseInt( img.id, 10 ) } // make id an integer explicitly
```